test/spec_resource_test.go

```diff
@@ -123,6 +124,57 @@ var _ = Describe("Setting Kubegres spec 'resource'", Label("group:5"), func() {
 		})
 	})
 
+	Context("GIVEN new Kubegres is created with spec 'sidecarContainer'", func() {
+
+		It("THEN created StatefulSets should have sidecarContainer set in pod template", func() {
+
+			log.Print("START OF: Test 'GIVEN new Kubegres is created with spec 'sidecarContainer")
+
+			containers := test.givenSidecarContainers("sidecarcontainer", "busybox")
+
+			test.givenNewKubegresSpecHasSidecarContainersSetTo(containers)
+
+			test.whenKubegresIsCreated()
+
+			test.thenStatefulSetStatesShouldHaveContainer("sidecarcontainer", "busybox")
+
+			test.dbQueryTestCases.ThenWeCanSqlQueryPrimaryDb()
+
+			test.keepCreatedResourcesForNextTest = true
+
+			log.Print("END OF: Test 'GIVEN new Kubegres is created with spec 'sidecarContainer")
+		})
+
+		It("THEN delete `sidecarContainer` field shouold remove the container form pod template spec", func() {
+			log.Print("START OF: Test 'THEN delete `sidecarContainer` field shouold remove the container form pod template spec")
+
+			test.givenExistingKubegresSpecSidecarContainersIsSetTo(nil)
+
+			test.whenKubernetesIsUpdated()
+
+			test.thenStatefulSetStatesShouldHaveNbreContainers(1)
+			test.dbQueryTestCases.ThenWeCanSqlQueryPrimaryDb()
+
+			test.keepCreatedResourcesForNextTest = true
+
+			log.Print("END OF: Test 'THEN delete `sidecarContainer` field shouold remove the container form pod template spec")
+		})
+
+		It("THEN add back `sidecarContainer` field should add the container back to pod template spec", func() {
+			log.Print("START OF: Test 'THEN add back `sidecarContainer` field should add the container back to pod template spec")
+
+			containers := test.givenSidecarContainers("sidecarcontainer", "busybox")
+			test.givenExistingKubegresSpecSidecarContainersIsSetTo(containers)
+
+			test.whenKubernetesIsUpdated()
+			test.thenStatefulSetStatesShouldHaveContainer("sidecarcontainer", "busybox")
+
+			test.dbQueryTestCases.ThenWeCanSqlQueryPrimaryDb()
+
+			log.Print("END OF: Test 'THEN add back `sidecarContainer` field should add the container back to pod template spec")
+		})
+	})
+
```

Comment:
it would be nice to have a test that modifies some properties of an existing sidecar container, like args or env vars.